scipy/stats/_discrete_distns.py

```diff
@@ -152,6 +153,80 @@ def _entropy(self, p):
 bernoulli = bernoulli_gen(b=1, name='bernoulli')
 
 
+class betabinom_gen(rv_discrete):
+    r"""A beta-binomial discrete random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability mass function for `betabinom` is:
+
+    .. math::
+
+       f(k) = \left(\begin{array}n \\ k\end{array}\right)
+            \frac{B(k + a, n - k + b)}{B(a, b)}
+
+    for ``k`` in ``{0, 1,..., n}``, where :math:`B(a, b)` is the beta function.
+
+    `betabinom` takes :math:`n`, :math:`a`, and :math:`b` as shape parameters.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Beta-binomial_distribution
+
+    %(after_notes)s
+
+    .. versionadded:: 1.4.0
+
+    %(example)s
+
+    """
+
+    def _rvs(self, n, a, b):
+        p = self._random_state.beta(a, b, self._size)
+        return self._random_state.binomial(n, p, self._size)
+
+    def _get_support(self, n, a, b):
+        return 0, n
+
+    def _argcheck(self, n, a, b):
+        return (n >= 0) & beta._argcheck(a, b)
```

Comment:
This is the only reason to import `beta` feom _continuous_distn, correct?
Then best spell out the restrictions on `a` and `b` and eliminate the import.

Reply:
Fixed.